pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -736,19 +736,24 @@ func (ssr *Snapshotter) resetFullSnapshotTimer() error {
 	return nil
 }
 
+// checkSnapstoreSecretUpdate checks if the snapstore secret has been updated
 func (ssr *Snapshotter) checkSnapstoreSecretUpdate() bool {
-	ssr.logger.Debug("checking the hash of snapstore secret...")
-	newSnapstoreSecretHash, err := snapstore.GetSnapstoreSecretHash(ssr.snapstoreConfig)
+	ssr.logger.Debug("checking the timestamp of snapstore secret...")
+	newSecretModTime, err := snapstore.GetSnapstoreSecretModTime(ssr.snapstoreConfig)
```

Comment:
```suggestion
	newSecretModTime, err := snapstore.GetSnapstoreSecretModTime(ssr.snapstoreConfig.Provider)
```
no need to pass whole snapStore config, provider string will be good enough.